package.json

```diff
@@ -2,15 +2,22 @@
   "name": "use-debounce",
   "version": "5.0.1",
   "description": "Debounce hook for react",
-  "main": "lib/index.js",
-  "module": "esm/index.js",
-  "types": "./lib",
+  "source": "src/index.ts",
+  "main": "dist/index.js",
+  "module": "dist/index.module.js",
+  "types": "dist/index.d.ts",
+  "exports": {
+    ".": {
+      "import": "./dist/index.module.js",
+      "require": "./dist/index.js"
+    }
+  },
   "sideEffects": false,
   "scripts": {
+    "size": "yarn build && size-limit",
     "test": "yarn jest",
-    "build": "yarn test && eslint \"src/**.ts\" && rm -rf ./lib/* ./esm/* && yarn build:cjs && yarn build:es",
-    "build:cjs": "tsc",
-    "build:es": "tsc -m esNext --outDir esm"
+    "build": "microbundle build -f es,cjs",
```

Comment:
`use-debounce` is a library that is included in many other projects. Each project decides which kind of source-maps it wants to use when it's better to use minification and so on. 

At the moment tsc util doesn't minify the code and it doesn't make source maps, however, microbundle does.

It leads to a number of problems for the library users:
1) they will have to include our source maps in the project
2) they will have to debug source maps instead of clean modules even in development mode

Microbundle is a good solution, but it doesn't help to `use-debounce` at all. The only profit it gives is the smaller size in synthetic benchmarks like bundlephobia. 




Let's leave tsc as it was. The comfort of users is more valuable than non-gzipped numbers.

Reply:
Also, microbundle produces only one file for each kind of build (instead of 2 files before). Which is a kind of deoptimization for some projects.

Reply:
Ok, let's go back to tsc then.

Btw. It's seems strange to me that your CJS bundle uses import/export. Microbundle produces require-based code and it's more generic thing for CJS/node AFAIK

Reply:
I've checked, CJS bundle uses `require`, as It should be:
```javascript
Object.defineProperty(exports, "__esModule", { value: true });
var react_1 = require("react");
var useDebouncedCallback_1 = require("./useDebouncedCallback");
function valueEquality(left, right) {
    return left === right;
}
```